lib/system/mm/malloc.nim

```diff
@@ -3,26 +3,53 @@
 
 proc allocImpl(size: Natural): pointer =
   result = c_malloc(size.csize_t)
-  when defined(zephyr):
+  when defined(zephyr) or defined(handleOOM):
     if result == nil:
       raiseOutOfMem()
 
 proc alloc0Impl(size: Natural): pointer =
   result = c_calloc(size.csize_t, 1)
-  when defined(zephyr):
+  when defined(zephyr) or defined(handleOOM):
     if result == nil:
       raiseOutOfMem()
 
 proc reallocImpl(p: pointer, newSize: Natural): pointer =
   result = c_realloc(p, newSize.csize_t)
-  when defined(zephyr):
+  when defined(zephyr) or defined(handleOOM):
     if result == nil:
       raiseOutOfMem()
 
+when defined(handleOOM):
+  proc nRealloc*(p: pointer, oldsize: csize_t, newsize: csize_t): pointer {.inline.} =
+    result = c_calloc(newSize.csize_t, sizeof(cchar).csize_t)
+    if result == nil:
+      raiseOutOfMem()
+    else:
+      if p != nil and newsize != 0 and newsize >= 0:
+        for i in 0..(newsize - 1):
+          if i <= oldSize:
+            cast[ptr UncheckedArray[cchar]](result)[i] = cast[ptr UncheckedArray[cchar]](p)[i]
+          else:
+            cast[ptr UncheckedArray[cchar]](result)[i] = '\0'
```

Comment:
What about `copyMem`?

Reply:
Not works unfortunately for unknown reason which I wasn't able to identify. Segfaults, looks like corrupted heap.

Reply:
Because you use `c_calloc`.